src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerNeo4jVersionValidationTest.java

```diff
@@ -0,0 +1,382 @@
+package org.neo4j.importer.v1;
+
+import org.junit.Test;
+import org.neo4j.importer.v1.distribution.Neo4jDistributions;
+import org.neo4j.importer.v1.validation.InvalidSpecificationException;
+
+import java.io.StringReader;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.neo4j.importer.v1.ImportSpecificationDeserializer.deserialize;
+
+public class ImportSpecificationDeserializerNeo4jVersionValidationTest {
+
+    @Test
+    public void fails_if_version_below_5_9_with_node_type_constraint() {
+        assertThatThrownBy(() -> deserialize(
+                        new StringReader(
+                                """
+            {
+                "version": "1",
+                "sources": [
+                    {
+                        "name": "a-source",
+                        "type": "bigquery",
+                        "query": "SELECT id, name FROM my.table"
+                    }
+                ],
+                "targets": {
+                    "nodes": [
+                        {
+                            "name": "a-node-target",
+                            "source": "a-source",
+                            "labels": ["Label1"],
+                            "properties": [
+                                {"source_field": "field_1", "target_property": "property1"}
+                            ],
+                            "schema": {
+                                "type_constraints": [
+                                    {"name": "type_constraint_1", "label": "Label1", "property": "property1"}
+                                ]
+                            }
+                        }
+                    ]
+                }
+            }
+            """
+                                        .stripIndent()),
+                        Neo4jDistributions.ENTERPRISE.of("5.0")))
+                .isInstanceOf(InvalidSpecificationException.class)
+                .hasMessageContainingAll(
+                        "1 error(s)",
+                        "0 warning(s)",
+                        "[$.targets.nodes[0].schema] [type_constraints], features are not supported by Neo4j 5.0 ENTERPRISE.");
```

Comment:
ideally, the message would look like this:

```suggestion
                        "[$.targets.nodes[0].schema] type constraints are not supported by Neo4j 5.0 ENTERPRISE.");
```